src/repr/src/row/encoding2.rs

```diff
@@ -1144,7 +1121,8 @@ impl Schema2<Row> for RelationDesc {
 /// A [`ColumnDecoder`] for a [`Row`].
 #[derive(Debug)]
 pub struct RowColumnarDecoder {
-    decoders: Vec<DatumColumnDecoder>,
+    len: usize,
+    decoders: Vec<(Arc<str>, Option<usize>, DatumColumnDecoder)>,
```

Comment:
I don't love having the null count here, IMO it feels like it should be a method on `DatumColumnDecoder` but that is a decent amount of typing so I'm okay with keeping it as-is! Do you mind adding a comment to this field though explaining what the elements of the tuple are?

Reply:
Yeah... I think to do this in a reasonable way I'd have to create a new `DatumDecoder` wrapper for `DatumColumnDecoder`, in the same sort of way we've had a `DatumEncoder` and `DatumColumnEncoder` split, where `DatumEncoder` handled the nullability info including stats. Totally reasonable, just some extra refactoring. I can take that as a followup if that sounds preferable!

Anyways will get some comments on these fields in the meantime.